### ISSUE TITLE
OOM after a few iterations on my model

### ISSUE THREAD

**Issue by tchaton** [PyTorchLightning/pytorch-lightning#9343]
Hello,

I have been trying to debug OOM after a few iterations on my model.
After investigation I saw that in loops/batch/training_batch_loop.py there is this piece of code (L235):
`result = self.training_step_and_backward(split_batch, batch_idx, opt_idx, optimizer, hiddens)`
`  if result is not None:`
`     return_result.update(result)`
`     return return_result.loss`
     
What I see is that return_result will keep the computation graph as it contains the loss. So I wonder what is the role of this variable? Also, where is the graph released, I could no go any further than "_training_step_and_backward_closure"? I don't understand why my model runs fine for a few iterations then there is some increase in memory.

_Originally posted by @Jovp in https://github.com/PyTorchLightning/pytorch-lightning/discussions/9332_

**Comment by awaelchli** [comment]
@Jovp @martenlienen 
Is #9336 attempting to fix this? If so, let's link the issue.

**Comment by Jovp** [comment]
That seems to be the case yes!
Julien

Le lun. 6 sept. 2021 à 13:32, Adrian Wälchli ***@***.***> a
écrit :
